service/s3/s3manager/upload.go

```diff
@@ -508,6 +508,7 @@ func (u *multiuploader) upload(firstBuf io.ReadSeeker) (*UploadOutput, error) {
 
 	// Read and queue the rest of the parts
 	for u.geterr() == nil {
+		num++
```

Comment:
What happens is that after the change it does one less iteration of the loop exiting at the first io.ErrUnexpectedEOF instead of iterating again and exiting at the io.EOF.

I moved it to make the unit tests pass!